portfolio/src/main/webapp/script.js

```diff
@@ -57,15 +57,67 @@ function getData() {
     printCom.innerHTML = '';
     comments.forEach((comment) => {
      printCom.appendChild(
-        createCommentFormat("Name: " + comment.username));
+        createElementFormat(("Name: " + comment.username)), 'li');
       printCom.appendChild(
-        createCommentFormat("Comment: " + comment.comment));
+        createElementFormat(("Comment: " + comment.comment)), 'li');
     })
   });
 }
 
-function createCommentFormat(text) {
-  const liElement = document.createElement('li');
-  liElement.innerText = text;
-  return liElement;
+function createElementFormat(text, elementType) {
+  const element = document.createElement(elementType);
+  element.innerText = text;
+  return element;
+}
+
+/**
+ * Display the user's login information
+ */
+function showLoginInfo() {
+  fetch('/login').then(response => response.json()).then((user) => {
+    const userInfo = document.getElementById('login-container');
+    userInfo.innerText = '';
+	
+    //display email and logout if logged in, otherwise login
+    if (user.logIn) {
```

Comment:
Do you actually need the 'logIn' field? Could you just not set the email if no one is logged in?

Reply:
I'm not 100% sure I understand exactly what you are asking but I used the 'logIn' field instead of just checking if the email is null just because I thought it would be easier and cleaner. Let me know if this is what you meant/want me to change it.

Reply:
I'd argue that having a null email is a lot cleaner and easier than having an extra field. You want to avoid having fields that provide you with the same information. A null email in the returned proto is a clear and understandable indication that there is no logged-in user, making 'login' unnecessary.